core/storage/src/lib.rs

```diff
@@ -122,6 +124,30 @@ macro_rules! ensure_get {
     }};
 }
 
+macro_rules! impl_get_keys_by_schemas_for {
+    ($self_: ident, $schema: ty, $vec: expr, $key_prefix: expr, $local_ibc_prefix: expr) => {{
+        let prepare_iter = $self_.adapter.prepare_iter::<$schema, _>($key_prefix)?;
+        let mut iter = prepare_iter.ref_to_iter();
+        let mut res = Vec::new();
+        loop {
+            let (key, _) = match iter.next() {
+                None => break,
+                Some(Ok(k_v_pair)) => k_v_pair,
+                Some(..) => continue,
+            };
+
+            let ibc_path: LocalIbcPath = key
+                .0
+                .try_into()
+                .map_err(|e| ProtocolError::new(ProtocolErrorKind::Ibc, Box::new(e)))?;
+            if ibc_path.is_contain($local_ibc_prefix) {
```

Comment:
Why not  replace `is_contain` with `starts_with` ?

Reply:
i saw sometimes user input parament like "channels/{}/cilentId/" , is that mean the function will match the key with "channels" and "clientId"?

Reply:
IMO, if it will iterator by prefix, should use `start with`.

Reply:
No, "channels/dddd/dfaf" will match, whereras "clientId/ddd**" won't.

Reply:
that`s right